### PR TITLE
To resolve #20

### LINKED ISSUE #20: Some stations do not report average daily temp
KSAF in Santa Fe, New Mexico does not record an average daily temperature, only the mean. In order for wunderground_parser.py to work in this case, all of the index values need to be shifted down by one after reading weather_data[0].

### PR DESCRIPTION
**Pull request by nyoung85**
Hey rhiever - cool work you’ve been doing.

For this issue I turned the weather_data value into a 2 dimensional list. First the html table is parsed by row and is stored in the first list dimension, then each of the rows is individually parsed to extract the variables. 

The “average mean temperature” that was throwing the Santa Fe data off now corresponds to weather_data[0][1], which doesn’t get evaluated in any city. That way, “actual max temperature” gets assigned from weather_data[1][0] across all cities.

Santa Fe also has no data for average and record precipitation, so I included conditionals to allow for these to be missing without raising the exception.

This is my first github pull request, btw — hope you find it helpful!

### CODE REVIEW

**Comment by rhiever** [comment]
Thanks @nyoung85! Going to run the scripts and see how it goes. Did you test this code on all of the stations in this repo?

**Comment by nyoung85** [comment]
Yep - all works

File: us-weather-history/wunderground_parser.py [hunk 1]
```diff
@@ -33,34 +33,53 @@ def parse_station(station):
...
 
-                    actual_precipitation = weather_data[9].text
+                    actual_precipitation = weather_data[4][0].text
                     if actual_precipitation == 'T':
-                        actual_precipitation = '0.0'
-                    average_precipitation = weather_data[10].text
-                    record_precipitation = weather_data[11].text
+                        actual_precipitation = '0.0' 
+
+                    #Test whether station collets average or record precipitation data
```
Comment (rhiever) [inline us-weather-history/wunderground_parser.py]: Typo: "collets"

File: us-weather-history/wunderground_parser.py [hunk 2]
```diff
@@ -33,34 +33,53 @@ def parse_station(station):
...
+                    if weather_data[4][1]:
+                        average_precipitation = weather_data[4][1].text
+                    else: average_precipitation = None
+                    if weather_data[4][2]:
+                        record_precipitation = weather_data[4][2].text
+                    else: record_precipitation = None
```
Comment (rhiever) [inline us-weather-history/wunderground_parser.py]: Please put the code on a separate line from the `else`.

File: us-weather-history/wunderground_parser.py [hunk 3]
```diff
@@ -33,34 +33,53 @@ def parse_station(station):
...
                     if actual_precipitation == 'T':
-                        actual_precipitation = '0.0'
-                    average_precipitation = weather_data[10].text
-                    record_precipitation = weather_data[11].text
+                        actual_precipitation = '0.0' 
+
+                    #Test whether station collets average or record precipitation data
+                    if weather_data[4][1]:
+                        average_precipitation = weather_data[4][1].text
+                    else: average_precipitation = None
```
Comment (rhiever) [inline us-weather-history/wunderground_parser.py]: Please put the code on a separate line from the `else`.

File: us-weather-history/wunderground_parser.py [hunk 4]
```diff
@@ -33,34 +33,53 @@ def parse_station(station):
...
+                        record_precipitation = weather_data[4][2].text
+                    else: record_precipitation = None
 
                     # Verify that the parsed data is valid
                     if (record_max_temp_year == '-1' or record_min_temp_year == '-1' or
                             int(record_max_temp) < max(int(actual_max_temp), int(average_max_temp)) or
                             int(record_min_temp) > min(int(actual_min_temp), int(average_min_temp)) or
-                            float(actual_precipitation) > float(record_precipitation) or
-                            float(average_precipitation) > float(record_precipitation)):
+                            ((record_precipitation is not None or average_precipitation is not None) and (float(actual_precipitation) > float(record_precipitation) or
```
Comment (rhiever) [inline us-weather-history/wunderground_parser.py]: Please split the statement following the `and` on the next line, so this line isn't so long.

File: us-weather-history/wunderground_parser.py [hunk 5]
```diff
@@ -68,14 +87,23 @@ def parse_station(station):
...
+
                     out_file.write('\n')
                     current_date += timedelta(days=1)
+                    
                 except:
                     # If the web page is formatted improperly, signal that the page may need
                     # to be downloaded again.
                     try_again = True
+                    
+                    '''#Used to produce a mock-up .csv for KSAF
```
Comment (rhiever) [inline us-weather-history/wunderground_parser.py]: Is this needed for the production code?

**Comment by rhiever** [comment]
Alrighty! I made some stylistic suggestions. Once we get those addressed, we should be in good shape to merge. Please ping me on here once you've committed those changes.

File: us-weather-history/wunderground_parser.py (hunk 5 above)
Comment (nyoung85) [inline us-weather-history/wunderground_parser.py]: No, it was just an easy way to make a KSAF csv file without having to manually enter the dates with incomplete/faulty data. I'll take it out this next pull request.

**Comment by nyoung85** [comment]
Changes made. I'm still new to github, though -- can you tell me if you got that last pull request?

**Comment by rhiever** [comment]
Everything you commit to the branch that you sent the PR from will automatically show up in the PR.

Tip for the future: When you fork someone's repo and want to make changes, create a new branch off of that fork to work in. That way you can work on multiple branches and potentially send multiple PRs from those branches without issue. (Repo maintainers won't always review/accept PRs quickly.)

Looks good. Thanks for the PR!

**Comment by nyoung85** [comment]
Thanks, good to know.

My pleasure!

**Pull request merged**